lib/lbt/resources/ResourcePool.js

```diff
@@ -177,14 +181,30 @@ class ResourcePool {
 	 * Retrieves the module info
 	 *
 	 * @param {string} name module name
+	 * @param {object} options
+	 * @param {boolean} options.preferDebugResources
+	 * @param {ResourceFilterList} options.debugBundleFilter
 	 * @returns {Promise<ModuleInfo>}
 	 */
-	async getModuleInfo(name) {
+	async getModuleInfo(name, options) {
 		let info = this._dependencyInfos.get(name);
 		if ( info == null ) {
 			// console.log("analyzing ", name);
+			let dbgResource;
+			if (options && options.preferDebugResources) {
+				// If requested, try to retrieve the corresponding debug-resource (xyz-dbg.js) and pass it to
+				// determineDependencyInfo as well. Its dependency analysis will perform better with the
+				// not-minified content of a resource.
+				const dbgName = ResourceInfoList.getDebugName(name);
+				if (dbgName && (!options.debugBundleFilter || !options.debugBundleFilter.matches(dbgName))) {
+					dbgResource = this._resourcesByName.get(dbgName);
```

Comment:
Maybe that's the best approach then? Rewriting the ResourcePool to tightly integrate with ui5-fs. Also, adding a new Class in ui5-fs to store metadata like the AST (or plain dependency information?) per-resource.

Every task modifying a resource should update this metadata or at least invalidate it in some way.

Then there would still be a final resource.json-generation task, but instead of analyzing resources it would use the metadata already available for most of them.